test/e2e/network/dns_configmap.go

```diff
@@ -180,7 +201,12 @@ func (t *dnsNameserverTest) run() {
 		"dnsmasq",
 		moreForeverTestTimeout)
 
-	t.c.CoreV1().ConfigMaps(t.ns).Delete(t.name, nil)
+	if t.name == "coredns" {
+		t.setConfigMap(&v1.ConfigMap{Data: originalConfigMapData})
+		t.deleteCoreDNSPods()
+	} else {
```

Comment:
we should keep this return to original, as there is a dependent test after it. 
But we still need the "defer .." restore to ensure it whatever the result of all tests.

Reply:
I would think that is also true for the kube-dns clear of ConfigMap.
Maybe be worth to create a function or restoring the original configmap configuration and defer that function on each test.

Reply:
Good point, defer seems like a better option.

Regarding kube-dns, the default kube-dns configmap is empty. So a reset would probably be enough. But having such function sounds good as well.